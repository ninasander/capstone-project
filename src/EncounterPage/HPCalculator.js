import React, { useState } from 'react'
import CounterButton from './CounterButton'
import styled from 'styled-components'

export default function HPCalculator({ setCurrentHP, currentHP, maxHP }) {
  const [input, setInput] = useState('')
  return (
    <CounterStyled>
      <CounterButton onClick={addToHP} operator="+" />
      <InputStyled
        min={0}
        value={input}
        onChange={(event) => setInput(event.target.value)}
        name="HPInput"
        type="number"
      />

      <CounterButton onClick={subtractFromHP} operator="-" />
    </CounterStyled>
  )

  function addToHP() {
    const calculatedHP = currentHP + Number(input)
    setCurrentHP(calculatedHP > maxHP ? maxHP : calculatedHP)
    setInput('')
  }
  function subtractFromHP() {
    const calculatedHP = currentHP - Number(input)
    setCurrentHP(calculatedHP < 0 ? 0 : calculatedHP)
    setInput('')
  }
}

const CounterStyled = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

const InputStyled = styled.input`
  width: 50px;
  position: relative;
`
