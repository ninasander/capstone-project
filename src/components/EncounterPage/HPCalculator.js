import React, { useState } from 'react'
import CalculatorButton from './CalculatorButton'
import styled from 'styled-components'

export default function HPCalculator({ setCurrentHP, currentHP, HP }) {
  const [hpInputValue, setHpInputValue] = useState('')
  return (
    <CounterStyled>
      <CalculatorButton onClick={addToHP} operator="Healing" />
      <InputStyled
        min={0}
        value={hpInputValue}
        onChange={(event) => handleChange(event)}
        name="HPInput"
        aria-label="HPInput"
        type="number"
        step="1"
      />

      <CalculatorButton onClick={subtractFromHP} operator="Damage" />
    </CounterStyled>
  )
  function handleChange(event) {
    const hpInputValue = event.target.value
    const regexp = new RegExp('^[0-9]+$')
    if (regexp.test(hpInputValue) || hpInputValue === '') {
      setHpInputValue(hpInputValue)
    }
  }

  function addToHP() {
    const calculatedHP = currentHP + Number(hpInputValue)
    setCurrentHP(calculatedHP > HP ? HP : calculatedHP)
    setHpInputValue('')
  }
  function subtractFromHP() {
    const calculatedHP = currentHP - Number(hpInputValue)
    setCurrentHP(calculatedHP < 0 ? 0 : calculatedHP)
    setHpInputValue('')
  }
}

const CounterStyled = styled.div`
  display: flex;
  grid-gap: 10px;
  align-items: center;
  justify-content: center;
`

const InputStyled = styled.input`
  width: 55px;
  height: 30px;
  text-align: center;
  position: relative;
  &::-webkit-outer-spin-button,
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  &[type='number'] {
    -moz-appearance: textfield;
  }
`
