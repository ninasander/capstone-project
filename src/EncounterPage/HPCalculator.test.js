import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import HPCalculator from './HPCalculator'
import { render } from '@testing-library/react'

describe('HPCalculator', () => {
  it('has an input field', () => {
    const { getByLabelText } = render(<HPCalculator />)
    const result = getByLabelText('HPInput')
    expect(result.length).toEqual(1)
  })
  it('has buttons', () => {
    const { getAllByRole } = render(<HPCalculator />)
    const result = getAllByRole('button')
    expect(result.length).toEqual(2)
  })
  it('renders correctly', () => {
    const tree = renderer.create(<HPCalculator />)
    expect(tree).toMatchSnapshot()
  })
})
