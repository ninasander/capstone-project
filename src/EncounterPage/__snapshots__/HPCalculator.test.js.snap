// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HPCalculator renders correctly 1`] = `
.c1 {
  font-size: 80%;
  border-style: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  text-align: center;
  background: green;
}

.c3 {
  font-size: 80%;
  border-style: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  text-align: center;
  background: red;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 50px;
  position: relative;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
    operator="+"
    type="button"
  >
    +
  </button>
  <input
    aria-label="HPInput"
    className="c2"
    min={0}
    name="HPInput"
    onChange={[Function]}
    step="1"
    type="number"
    value=""
  />
  <button
    className="c3"
    onClick={[Function]}
    operator="-"
    type="button"
  >
    -
  </button>
</div>
`;
